### PR TITLE
(#668) New export functionality

### PR DESCRIPTION
**Pull request by squaregoldfish**
The old export functionality has been replaced with a new bean and related bits and pieces.

### CODE REVIEW

**Comment by squaregoldfish** [comment]
Also fixes #376

File: WebApp/src/uk/ac/exeter/QuinCe/data/Dataset/DataSetDB.java [hunk 1]
```diff
@@ -31,8 +32,14 @@
...
    */
   private static final String GET_DATASETS_QUERY = "SELECT "
-      + "id, instrument_id, name, start, end, status, properties, last_touched "
-        + "FROM dataset WHERE instrument_id = ? ORDER BY start ASC";
+      + "d.id, d.instrument_id, d.name, d.start, d.end, d.status, "
+      + "d.properties, d.last_touched, COUNT(c.user_flag) "
+      + "FROM dataset d "
+      + "LEFT JOIN dataset_data dd ON d.id = dd.dataset_id "
+      + "LEFT JOIN equilibrator_pco2 c ON c.measurement_id = dd.id AND c.user_flag = " + Flag.VALUE_NEEDED + " "
+      + "WHERE instrument_id = ? "
```
Comment (jonasfh) [inline WebApp/src/uk/ac/exeter/QuinCe/data/Dataset/DataSetDB.java]: WHERE d.instrument_id = ?

File: WebApp/src/uk/ac/exeter/QuinCe/data/Dataset/DataSetDB.java [hunk 2]
```diff
@@ -31,8 +32,14 @@
...
+      + "GROUP BY d.id "
+      + "ORDER BY start ASC";
```
Comment (jonasfh) [inline WebApp/src/uk/ac/exeter/QuinCe/data/Dataset/DataSetDB.java]: ORDER BY d.start ASC

File: WebApp/src/uk/ac/exeter/QuinCe/web/system/ResourceManager.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ public void contextInitialized(ServletContextEvent event) {
         }
 
         try {
-            String filePath = (String) servletContext.getInitParameter("configuration.path");
```
Comment (jonasfh) [inline WebApp/src/uk/ac/exeter/QuinCe/web/system/ResourceManager.java]: This section has 4 spaces indenting?

**Pull request merged**